docs/MatrixOne/Reference/Data-Types/0.8-enum-type.md

```diff
@@ -0,0 +1,111 @@
+
+```sql
+CREATE TABLE table_name (
+    ...
+    col ENUM ('value1','value2','value3'),
+    ...
+);
+```
+
+### Explanations
+
+- `ENUM` is a keyword used to declare an enumeration type.
+- value1 to valuen is the optional list of choices for this `ENUM` type. The value of a column using the `ENUM` type can only be one of the values listed above.
+- Enumeration values can be of type string, int, or time.
+
+__Note:__ You can have multiple enumeration values in the `ENUM` data type. However, it is recommended to keep the number of enumeration values below 20.
+
+## Example Explanation
+
+The value of an `ENUM` type must be selected from a predefined list of values. The following example will help you understand:
+
+```sql
+CREATE TABLE enumtable (
+    id INT NOT NULL AUTO_INCREMENT,
+    color ENUM('red', 'green', 'blue'),
+    PRIMARY KEY (id)
+);
+```
+
+The above statement will create a table named `enumtable`, which contains an enum type field named `color`. The value of the `color` field must be one of `red`, `green`, or `blue`. At the same time, according to the order of column definition, the indexes of `red`, `green`, and `blue` are 1, 2, and 3, respectively.
+
+### Insert ENUM values
+
+When inserting data into a field of an enumeration type, only predefined enumeration values ​​or `NULL` can be inserted. An error is raised if the inserted value is not in the predefined list. For example:
+
+```sql
+INSERT INTO enumtable (id, color) VALUES ('01', 'red');
+-- 'red' is in the predefined list; the insertion was successful
+INSERT INTO enumtable (id, color) VALUES ('02', 'yellow');
+-- 'yellow' is not in the predefined list, an error will be generated
+INSERT INTO enumtable (id, color) VALUES ('03', NULL);
+-- The enumeration member does not define not null; the insertion is successful
+```
+
+In addition to enumeration values, data can be inserted into ENUM columns using numeric indexes on enumeration members. For example:
+
+```sql
+INSERT INTO enumtable (id, color) VALUES ('04', 2);
+-- Since the index of `green` is 2, this data is successfully inserted
+```
+
+- **NON-NULL CONSTRAINT FOR ENUM RESTRICTIONS**
+
+If we defined the `color` column `NOT NULL` when creating the table:
+
+```sql
+CREATE TABLE enumtable (
+    id INT NOT NULL AUTO_INCREMENT,
+    color ENUM('red', 'green', 'blue') NOT NULL,
+    PRIMARY KEY (id)
+);
+```
+
+When inserting a new row without specifying a value for the color column, MatrixOne will use the first enumeration member as the default value:
+
+```sql
+INSERT INTO enumtable (id) VALUES ('05');
+-- Here, the first enumeration member `red` will be assigned as the default value for the column with id 05
+```
+
+In non-strict SQL mode, if an invalid value is inserted into an ENUM column, MatrixOne will insert an empty string '' with a numeric index of 0. If strict SQL mode is enabled, attempting to insert an invalid `ENUM` value will result in an error.
```

Comment:
What's strict and non-strict SQL mode in MatrixOne? this is ambiguous

Reply:
strict and non-strict SQL mode only implement syntax support, it's not supported here.  I'll delete this description